handlers/external.js

```diff
@@ -41,57 +41,23 @@ module.exports = {
     * @param {fn} cb
     *        a node style callback(err, results) to send data when job is finished
     */
-   fn: function handler(req, cb) {
+   fn: async function handler(req, cb) {
       req.log("in process_manager.external");
 
-      // get the AB for the current tenant
-      ABBootstrap.init(req)
-         .then((AB) => {
-            var user = req.param("user");
-            var uuid = req.param("uuid");
-            var response = req.param("response");
+      try {
+         // get the AB for the current tenant
+         const AB = await ABBootstrap.init(req);
+         const uuid = req.param("uuid");
+         const roles = await AB.objectRole().model().findAll();
 
-            req.retry(() =>
-               AB.objectProcessForm().model().update(uuid, {
```

Comment:
It would be more efficient to leave the `.update()` here in this service handler.  That we we only perform the update 1 time.